src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -467,7 +472,7 @@ void MergeTreeData::checkProperties(
                                 "Primary key must be a prefix of the sorting key, "
                                 "but the column in the position {} is {}", i, sorting_key_column +", not " + pk_column);
 
-            if (!primary_key_columns_set.emplace(pk_column).second)
+            if (!primary_key_columns_set.emplace(pk_column).second && !allow_suspicious_indices && !attach)
```

Comment:
So there seems to exist some duplication check already (l. 476). Not sure - can the existing check somehow be consolidated into the new check?